napalm_nxos/nxos.py

```diff
@@ -169,19 +234,128 @@ def _get_reply_table(self, result, table_name, row_name):
 
     def _get_command_table(self, command, table_name, row_name):
-        json_output = self.device.show(command)
+        if self.ssh_connection:
+            command += ' | json'
+            try:
+                json_output = json.loads(self.device.send_command(command))
+            except ValueError:
+                return {}
+        else:
+            json_output = self.device.show(command)
         return self._get_reply_table(json_output, table_name, row_name)
 
     def is_alive(self):
-        if self.device:
-            return {'is_alive': True}
+        if self.ssh_connection:
+            return {
+                'is_alive': self.device.remote_conn.transport.is_active()
+                }
         else:
-            return {'is_alive': False}
+            if self.device:
+                return {'is_alive': True}
+            else:
+                return {'is_alive': False}
 
     def load_replace_candidate(self, filename=None, config=None):
+        if self.ssh_connection:
+            self._replace_candidate_ssh(filename, config)
+        else:
+            self._replace_candidate(filename, config)
         self.replace = True
         self.loaded = True
 
+    def _get_flash_size(self):
+        command = 'dir {}'.format('bootflash:')
+        output = self.device.send_command(command)
+
+        match = re.search(r'(\d+) bytes free', output)
+        bytes_free = match.group(1)
+
+        return int(bytes_free)
+
+    def _enough_space(self, filename):
+        flash_size = self._get_flash_size()
+        file_size = os.path.getsize(filename)
+        if file_size > flash_size:
+            return False
+        return True
+
+    def _verify_remote_file_exists(self, dst, file_system='bootflash:'):
+        command = 'dir {0}/{1}'.format(file_system, dst)
+        response = self.device.send_command(command)
+        if 'No such file' in response:
+            raise ReplaceConfigException('Could not transfer file.')
+
+    def _replace_candidate_ssh(self, filename, config):
+        if not filename:
+            file_content = self.fix_checkpoint_string(config, self.replace_file)
+            filename = self._create_tmp_file(config)
+        else:
+            if not os.path.isfile(filename):
+                raise ReplaceConfigException("File {} not found".format(filename))
+
+        self.replace_file = filename
+        with open(self.replace_file, 'r+') as f:
+            file_content = f.read()
+            file_content = self.fix_checkpoint_string(file_content, self.replace_file)
+            f.write(file_content)
+
+        if not self._enough_space(self.replace_file):
+            msg = 'Could not transfer file. Not enough space on device.'
+            raise ReplaceConfigException(msg)
+
+        self._check_file_exists(self.replace_file)
+        dest = os.path.basename(self.replace_file)
+        full_remote_path = 'bootflash:{}'.format(dest)
+        with paramiko.SSHClient() as ssh:
```

Comment:
Can we use Netmiko's scp_handler.py here. It will be more robust as it will support all Netmiko's arguments (for example, SSH keys and not just username/password).